### DISCUSSION TITLE
Textures are pixelated in google scene viewer

### DISCUSSION THREAD

**Discussion by AksAman** [google/model-viewer#1995] [Q&A]
### Description

Hi, i've been using model-viewer to display models in google scene viewer ar, but sometimes, some textures are pixelated in thr ar view but looks fine in the online scene viewers (like https://arvr.google.com/scene-viewer-preview). 

#### Live Demo
https://dybo-io.github.io/webar_customizer/?id=GITHUB_ISSUE

### Version
https://unpkg.com/@google/model-viewer@1.4.1/dist/model-viewer.min.js

### Pictures
This is the pixelation in arview
![WhatsApp Image 2021-02-01 at 22 37 47](https://user-images.githubusercontent.com/28787976/106493726-89f5bd00-64df-11eb-8d84-d2de2366e13c.jpeg)
In Online scene viewer, there is no pixelation
![no_pixelation](https://user-images.githubusercontent.com/28787976/106493740-8feb9e00-64df-11eb-89e2-98fec55bc7a0.png)

### Browser Affected

- [x] Chrome (v 88.0.4324.93)
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android (Nokia 6.1 Plus, Android 10)
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
Those look equally pixelated to me, just one is more zoomed-in. Any model will look pixelated if you get close enough to it.

**Comment by AksAman** [comment]
@elalish I am uploading the images again, with similar zoom level.

Screenshot from AR Scene viewer from my phone
You can see that the leather pattern is completely pixelated.
![106493726-89f5bd00-64df-11eb-8d84-d2de2366e13c](https://user-images.githubusercontent.com/28787976/106567655-2c518700-6558-11eb-978c-90392d6daac0.jpeg)

Screenshot from https://arvr.google.com/scene-viewer-preview
(No pixelation) (The leather pattern is atleast visible)
![no_pixelation](https://user-images.githubusercontent.com/28787976/106567666-32dffe80-6558-11eb-80bb-850fda9cb809.png)

**Reply by elalish** [reply]
Thanks, I see what you mean now. Of course SceneViewer is a separate project from model-viewer, but @tpsiaki can take a look at that. From you demo it appears the result is rendered correctly by model-viewer (at least once I resize it to be visible). Therefore WebXR mode should render it just fine as well; have you tried that?

**Reply by AksAman** [reply]
Yeah, webxr works correctly, but lags on my phone ( (Nokia 6.1 Plus, Android 10). But, i am not opting for webxr right now, because it's not supported by iOS.

**Reply by elalish** [reply]
SceneViewer is also not supported by iOS; iOS only uses QuickLook, and I think you'll find more differences between QuickLook and SceneViewer than between SceneViewer and WebXR. But yes, the world will be much improved when Safari gets WebXR support and we can finally have a cross-platform AR solution.

**Reply by AksAman** [reply]
I understand that pixelation is resolved in webXR, but as i said webXR lags a lot which isn't a good user experience. Also, i've noticed that a couple of companies have used scene viewer without any pixelation. So, i'm trying to understand what exactly is causing pixelation in our pipeline and how to avoid it!

**Reply by elalish** [reply]
Fair, I'll have to defer to @tpsiaki for SceneViewer. We are working on the lag issue; should be fixed in an upcoming version of Chrome.

**Reply by tpsiaki** [reply]
@AksAman The link shared above doesn't seem to launch to a model anymore. Do you have a link to the model being used. The most likely reason for this would be if you have very large textures. In Scene Viewer we recommend textures no larger than [2Kx2K](https://developers.google.com/ar/develop/java/scene-viewer#file_requirements_for_models) and when we load models if you have larger textures they will get downscaled as part of decoding/loading them into memory.

**Reply by AksAman** [reply]
@tpsiaki I checked the link and it's working. This page will show up and the user needs to click on "View in AR" Button
<img src="https://user-images.githubusercontent.com/28787976/107394846-4b38b600-6b22-11eb-99bc-b6a3e4487a27.png" width="200"> 

Anyways i have uploaded the glb file here 
[Model.zip](https://github.com/google/model-viewer/files/5952772/GITHUB_ISSUE_1.zip)

I am also uploading a video recording from scene viewer, showing the pixelation issue on the right arm.

https://user-images.githubusercontent.com/28787976/107396576-e67e5b00-6b23-11eb-826d-81c678daead7.mp4

**Reply by tpsiaki** [reply]
Thanks for sharing the model.  We've reproed it and can confirm that this is an issue with Scene Viewer.  We'll work on getting a fix out.

**Reply by AksAman** [reply]
Hey @tpsiaki, Any updates regarding the scene viewer issue?
